### ISSUE TITLE
Inaccurate documentation of OGRCoordinateTransformation transform methods

### ISSUE THREAD

**Issue by SunBlack** [OSGeo/gdal#9069]
The documentation of the class `OGRCoordinateTransformation` describes the return value of the transformation methods as:
https://github.com/OSGeo/gdal/blob/4daa3198862100f27a9180a7400e294c8c01fcdc/ogr/ogr_spatialref.h#L856-L857

This is basically only partially true, by checking the code. The underlying method contains only a single occurrence with `return FALSE;` - all others are `return TRUE;`:

https://github.com/OSGeo/gdal/blob/4daa3198862100f27a9180a7400e294c8c01fcdc/ogr/ogrct.cpp#L2257-L2862

In other words, it is incorrect that the method returns `FALSE` if no coordinate could be transformed. If, for example, all coordinates are `PROJ_ERR_COORD_TRANSFM_OUTSIDE_PROJECTION_DOMAIN`, the method still returns `TRUE`.

Correct would be sth like this (as the result is independent of the success of the transformation of individual coordinates):
```
TRUE if a transformation could be found, otherwise FALSE.
```
